mantle/cmd/kola/switchkernel.go

```diff
@@ -0,0 +1,249 @@
+
+	c, err := flight.NewCluster(&platform.RuntimeConfig{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to create new cluster")
+	}
+	defer c.Destroy()
+
+	m, err := c.NewMachine(userdata)
+	if err != nil {
+		return errors.Wrapf(err, "failed to spawn new machine")
+	}
+	defer m.Destroy()
+
+	err = testSwitchKernel(c)
+	if err != nil {
+		return errors.Wrapf(err, "failed switch kernel test")
+	}
+
+	return nil
+}
+
+// Drops Kernel RT RPM files under the directory `localPath` to `$homeDir/kernel-rt-rpms` directory in m
+func dropRpmFilesAll(m platform.Machine, localPath string) error {
+	fmt.Println("Dropping RT Kernel RPMs...")
+	re := regexp.MustCompile(`^kernel-rt-.*\.rpm$`)
+	files, err := ioutil.ReadDir(localPath)
+	if err != nil {
+		return err
+	}
+	for _, f := range files {
+		filename := f.Name()
+		filepath := strings.TrimSuffix(localPath, "/") + "/" + filename
+		targetPath := fmt.Sprintf("%s/kernel-rt-rpms/%s", homeDir, filename)
+		if re.MatchString(filename) {
+			in, err := os.Open(filepath)
+			if err != nil {
+				return err
+			}
+			if err := platform.InstallFile(in, m, targetPath); err != nil {
+				return errors.Wrapf(err, "failed to upload %s", filename)
+			}
+		}
+	}
+	return nil
+}
+
+func switchDefaultToRtKernel(c platform.Cluster, m platform.Machine) error {
+	// run the script to switch from default kernel to rt kernel
+	fmt.Println("Switching from Default to RT Kernel...")
+	cmd := "sudo " + homeDir + "/switch-kernel.sh default rt-kernel " + homeDir + "/kernel-rt-rpms/"
+	stdout, stderr, err := m.SSH(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to run %s", cmd)
+	}
+	fmt.Println(fmt.Sprintf("%s", stderr))
+	fmt.Println(fmt.Sprintf("%s", stdout))
+
+	// reboot the machine to switch kernel
+	fmt.Println("Rebooting machine...")
+	err = m.Reboot()
+	if err != nil {
+		return errors.Wrapf(err, "failed to reboot machine")
+	}
+
+	// check if the kernel has switched to rt kernel
+	fmt.Println("Checking kernel type...")
+	cmd = "uname -v | grep -q 'PREEMPT RT'"
+	_, _, err = m.SSH(cmd)
+	if err == nil {
+		return errors.Wrapf(err, "failed to run %s", cmd)
+	}
+	fmt.Println("Switched to RT Kernel successfully!")
+
+	return nil
+}
+
+func switchRtKernelToDefault(c platform.Cluster, m platform.Machine) error {
+	// run the script to switch from rt kernel to default
+	fmt.Println("Switching from RT to Default Kernel...")
+	cmd := "sudo " + homeDir + "/switch-kernel.sh rt-kernel default"
+	stdout, stderr, err := m.SSH(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to run %", cmd)
+	}
+	fmt.Println(fmt.Sprintf("%s", stderr))
+	fmt.Println(fmt.Sprintf("%s", stdout))
+
+	// reboot the machine to switch kernel
+	fmt.Println("Rebooting machine...")
+	err = m.Reboot()
+	if err != nil {
+		return errors.Wrapf(err, "failed to reboot machine")
+	}
+
+	// check if the kernel has switched back to default kernel
+	fmt.Println("Checking kernel type...")
+	cmd = "! $(uname -v | grep -q 'PREEMPT RT')"
```

Comment:
I think it's better to use `grep -v` then `! grep` - it more clearly expresses intent (and ensures that you still *do* get an error if e.g. an input file doesn't exist or grep got OOM killed or whatever).

And see above too about the `_`.

Reply:
Updated with `grep -v` and tested on a RHCOS machine, worked without issue